### PR TITLE
[Bugfix] Fix gridDim.y overflow for large row counts

### LINKED ISSUE #45099: [Bug]:  [DeepSeek-V4-Flash][MTP] CUDA invalid argument during profile_run with DP4 + EP + MegaMoE on B_00
### Your current environment

<details>
 main branch 
</details>

### 🐛 Describe the bug

DeepSeek-V4-Flash can start successfully without MTP using DP4 + EP + MegaMoE.

However, when enabling MTP speculative decoding, the server crashes during startup/profile_run with:

CUDA error: invalid argument

This happens with both:

{"method":"mtp","num_speculative_tokens":1}

and

{"method":"mtp","num_speculative_tokens":3}

The crash happens before serving requests.

Command
```
CUDA_VISIBLE_DEVICES=0,5,6,7 vllm serve /root/model/DeepSeek-V4-Flash \
  --served-model-name deepseek-v4-flash \
  --trust-remote-code \
  --kv-cache-dtype fp8 \
  --block-size 256 \
  --data-parallel-size 4 \
  --enable-expert-parallel \
  --moe-backend deep_gemm_mega_moe \
  --gpu-memory-utilization 0.92 \
  --max-model-len 8192 \
  --max-num-seqs 128 \
  --max-num-batched-tokens 32768 \
  --compilation-config '{"cudagraph_mode":"FULL_AND_PIECEWISE","custom_ops":["all"]}' \
  --attention_config.use_fp4_indexer_cache True \
  --speculative-config '{"method":"mtp","num_speculative_tokens":1}' \
  --tokenizer-mode deepseek_v4
```
see detail 

https://paste.ubuntu.com/p/ZK55nz9d3r/

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by JasonLi314**
## Purpose

Fixes [#45099](https://github.com/vllm-project/vllm/issues/45099). Prior code mixed up cuda gridDim.x and gridDim.y, mapping the row dimension (mn) to gridDim.y, which CUDA caps at 65535. This fix:

* Assigns row dimension mn to gridDim.x instead of gridDim.y, as grid x limit is capped at 2^31 - 1.
* Fixed the launch guard, which previously checked both grid dims against `INT32_MAX`. Now grid x is checked at `INT32_MAX` while grid y is checked against 65535.

This bug doesn't surface in most cases as mn is often small, but this edge case can be hit by the DeepSeek-V4 MTP draft path in the issue above, which quantizes a 3D residual so mn = tokens * hc_mult and exceeds 65535 during profile run.

## Test Plan

Added new unit test, able to reproduce error `CUDA error: invalid configuration argument (cudaErrorInvalidConfiguration)` in unit test without the fix.

Notice: unit test was able to reproduce `CUDA error: invalid configuration argument` which is a slightly different err message than the one from profile run `CUDA error: invalid argument`. This is most likely caused by different cuda versions, but regardless both errors come from gridDim.y being too large at kernel launch.

##### Running new unit test without the fix

```bash
$ pytest tests/kernels/quantization/test_per_token_group_quant.py -v --tb=short
...

====================================================== FAILURES ======================================================
___________________________________ test_per_token_group_quant_fp8_packed_large_mn ___________________________________
tests/kernels/quantization/test_per_token_group_quant.py:376: in test_per_token_group_quant_fp8_packed_large_mn
    ref_q, ref_s = fp8_utils.per_token_group_quant_fp8(x, group_size, use_ue8m0=True)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
vllm/model_executor/layers/quantization/utils/fp8_utils.py:574: in per_token_group_quant_fp8
    x_s = torch.empty(shape, device=x.device, dtype=torch.float32)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E   torch.AcceleratorError: CUDA error: invalid configuration argument
E   Search for `cudaErrorInvalidConfiguration' in https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html for more information.
E   CUDA kernel errors might be asynchronously reported at some other API call, so the stacktrace below might be incorrect.
E   For debugging consider passing CUDA_LAUNCH_BLOCKING=1
E   Compile with `TORCH_USE_CUDA_DSA` to enable device-side assertions.

...

============================================== short test summary info ===============================================
FAILED tests/kernels/quantization/test_per_token_group_quant.py::test_per_token_group_quant_fp8_packed_large_mn - torch.AcceleratorError: CUDA error: invalid configuration argument
==================================== 1 failed, 119 passed, 16 warnings in 27.20s =====================================
```

---

##### Running new unit test with the fix

```bash
$ pytest tests/kernels/quantization/test_per_token_group_quant.py -v --tb=short
...
tests/kernels/quantization/test_per_token_group_quant.py::test_per_token_group_quant_fp8_packed_large_mn PASSED [ 95%]
...
========================================= 120 passed, 16 warnings in 26.80s ==========================================
```

---

<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [X] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [NA] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.

</details>

### CODE REVIEW

**Comment by JasonLi314** [comment]
PR is ready for review. Please take a look when available. Thanks!

File: tests/kernels/quantization/test_per_token_group_quant.py [hunk 1]
```diff
@@ -345,6 +345,63 @@ def test_per_token_group_quant_fp8_packed_zero_fills_padded_output_q(
         )
 
 
+@pytest.mark.skipif(
+    not current_platform.is_cuda_alike(),
+    reason="DeepGEMM not available on this platform",
```
Comment (tlrmchlsmth) [inline tests/kernels/quantization/test_per_token_group_quant.py]: nit: could improve this reason str (but I see that it is consistent with other `reason`s in this file)

**Review by tlrmchlsmth — APPROVED** [review]
Thanks for the fix!

**Comment by JasonLi314** [comment]
The earlier CI failures came from unrelated upstream issues on a stale branch, not from this fix. Rebased to latest commit. Re-compiled and ran unit test again after rebase.

This PR only changes a CUDA quantization kernel, while the failed CI runs hit an API server startup bug already tracked in issues #45596 and #45597.

Also improved pytest `reason` field based on comment above.

**Comment by JasonLi314** [comment]
Hi @tlrmchlsmth, thanks for the review. Quick CI update:

This PR only touches quantization, and quantization & kernel checks have passed. There are couple CI failures that are unrelated to this change:

**Infrastructure**
- arm CPU Docker build failed pulling a PyTorch dependency (CDN returned http 503)
- one CPU multimodal job timed out

**Upstream dependency (NIXL)**

five NIXL PD jobs fail at startup because the installed NIXL package is missing its native extension, probably because NIXL 1.13.0 was just released. Couple similar NIXL related issues:
  - prior fix when NIXL packaging last broke PD CI: https://github.com/vllm-project/vllm/pull/39851
  - later bump to newer NIXL versions: https://github.com/vllm-project/vllm/pull/42364
  - open follow-up to tighten version constraints: https://github.com/vllm-project/vllm/pull/44143

**Main branch (unrelated)**
- spec decode + LoRA job hit a GPU crash in an unrelated LoRA kernel; the FP8 parts of that same job passed
- one e2e test failed on a whitespace-only output mismatch

I do not think rerunning will help here. Could you force merge when you have a moment? Thanks!

**Comment by JasonLi314** [comment]
Need to force merge.

2 CI failed and neither of them are related to this change.

* **extract-hidden-states-integration-2-gpus**: test_extract_hidden_states_tp2 timed out at 60s. Log shows a HuggingFace 429 at startup eating 27s, so the TP2 engine was still booting when pytest killed it. This is KV connector / speculative decoding.
* **cudagraph**: test_cudagraph_compilation_combo[FA2-FULL-0-True] failed because GPU memory didn't drop after teardown (stuck ~10.7 GiB for 120s). CUDA graph + FA2 integration test.

**Comment by JasonLi314** [comment]
All tests passed. Please merge thx.
